common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -418,6 +415,47 @@ static boolean equals(byte[] bytes1, int startPos1, byte[] bytes2, int startPos2
         }
     }
 
+    static int equalsConstantTime(byte[] bytes1, int startPos1, byte[] bytes2, int startPos2, int length) {
+        long result = 0;
+        final long baseOffset1 = BYTE_ARRAY_BASE_OFFSET + startPos1;
+        final long baseOffset2 = BYTE_ARRAY_BASE_OFFSET + startPos2;
+        final int remainingBytes = length & 7;
+        final long end = baseOffset1 + remainingBytes;
+        for (long i = baseOffset1 - 8 + length, j = baseOffset2 - 8 + length; i >= end; i -= 8, j -= 8) {
+            result |= UNSAFE.getLong(bytes1, i) ^ UNSAFE.getLong(bytes2, j);
```

Comment:
@carl-mastrangelo - The following was collapsed but still open ...

> Also, how are you sure that byte arrays are word aligned?

`PlatformDependent.equalsConstantTime` checks if aligned access is allowed in general, but no check is done on the individual arrays. Do you know how I would check, and an example allocation which I can use to verify this method?

> And fyi, I am proposing a similar change for Go: https://go-review.googlesource.com/#/c/24422/1/src/crypto/subtle/constant_time.go

I guess this would depend upon the answer to the first question ... maybe investigate this as a followup?